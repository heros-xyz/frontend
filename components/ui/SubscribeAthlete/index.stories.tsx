import type { Meta, StoryObj } from "@storybook/react";
import SubscribeAthlete from ".";

const meta: Meta<typeof SubscribeAthlete> = {
  title: "Components/SubscribeAthlete",
  component: SubscribeAthlete,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
  argTypes: {},
};

export default meta;
type Story = StoryObj<typeof SubscribeAthlete>;

export const SubscribeAthlete1: Story = {
  args: {
    message:
      "This action cannot be undone. Are you sure you want to delete this interaction?",
    confirm: "yes, delete Interaction",
    cancel: "Cancel",
    onCancel: () => console.log("cancel"),
    onSubmit: () => console.log("submit"),
  },
};
