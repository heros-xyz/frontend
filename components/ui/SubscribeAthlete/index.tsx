import { Box, Button, Center, Link, Text } from "@chakra-ui/react";
import React from "react";
import NextLink from "next/link";
import { useRouter } from "next/router";
import { AlertIcon } from "@/components/svg/Alert";

const SubscribeAthlete = () => {
  const router = useRouter();

  return (
    <Box bg="white" minH={50} borderRadius={8} zIndex="1" pb={6}>
      <Center mb={3}>
        <Center
          bg="#FEE2E2"
          w={{ base: "12", xl: "20" }}
          h={{ base: "12", xl: "20" }}
          borderRadius="full"
        >
          <AlertIcon
            w={{ base: "20px", xl: "30px" }}
            h={{ base: "18px", xl: "26px" }}
          />
        </Center>
      </Center>
      <Center mb={{ base: 4, lg: 6 }}>
        <Text
          fontSize={{ base: "lg", lg: "2xl" }}
          fontWeight={{ base: 400, lg: 500 }}
        >
          You have to sign in to subscribe.
        </Text>
      </Center>
      <Center mb={{ base: 4, lg: 6 }}>
        <Button
          w={{ base: "full", lg: "fit-content" }}
          variant="primary"
          size="lg"
          onClick={() => {
            router.push({
              pathname: "/sign-in",
            });
          }}
        >
          Sign In
        </Button>
      </Center>
      <Center>
        <Text mr={2} color="primary" fontWeight={500}>
          Or
        </Text>{" "}
        <Link
          as={NextLink}
          href={`/joining-as`}
          color="secondary"
          textDecoration="underline"
        >
          Sign up here
        </Link>
      </Center>
    </Box>
  );
};

export default SubscribeAthlete;
