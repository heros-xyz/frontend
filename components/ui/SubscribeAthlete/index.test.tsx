import { render } from "@testing-library/react";
import mockRouter from "next-router-mock";
import SubscribeAthlete from ".";
jest.mock("next/router", () => require("next-router-mock"));

it("renders SubscribeAthlete unchanged", () => {
  mockRouter.push("/initial-path");
  const { container } = render(<SubscribeAthlete />);
  expect(container).toMatchSnapshot();
});
