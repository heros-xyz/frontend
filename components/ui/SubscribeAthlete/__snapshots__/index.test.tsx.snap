// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SubscribeAthlete unchanged 1`] = `
<div>
  <div
    class="css-emu7uy"
  >
    <div
      class="css-f78n2q"
    >
      <div
        class="css-1xq4rq0"
      >
        <svg
          class="chakra-icon css-1j5z9d7"
          focusable="false"
          viewBox="0 0 20 18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.99955 7V9M9.99955 13H10.0095M3.07134 17H16.9277C18.4673 17 19.4296 15.3333 18.6598 14L11.7316 2C10.9618 0.666667 9.0373 0.666667 8.2675 2L1.33929 14C0.569492 15.3333 1.53174 17 3.07134 17Z"
            stroke="#FF6767"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
    <div
      class="css-1mfso1b"
    >
      <p
        class="chakra-text css-j4z01g"
      >
        You have to sign in to subscribe.
      </p>
    </div>
    <div
      class="css-1mfso1b"
    >
      <button
        class="chakra-button css-1hh46d3"
        type="button"
      >
        Sign In
      </button>
    </div>
    <div
      class="css-gmuwbf"
    >
      <p
        class="chakra-text css-1u16guu"
      >
        Or
      </p>
       
      <a
        class="chakra-link css-18noweo"
        href="/joining-as"
      >
        Sign up here
      </a>
    </div>
  </div>
</div>
`;
